src/dist/replication/lib/replica_learn.cpp

```diff
@@ -1416,14 +1558,37 @@ error_code replica::apply_learned_state_from_private_log(learn_state &state)
                                },
                                offset);
 
+    // update first_learn_start_decree, the position where the first round of LT_LOG starts from.
+    // we use this value to determine whether to learn back from min_confirmed_decree
+    // for duplication:
+    //
+    //                confirmed
+    //                    |
+    // learner's plog: ==[=========[--------------]
+    //                   |         |              |
+    //                   |       gced           committed
+    //     first_learn_start_decree
+    //
+    // because the learned logs (under `learn/` dir) have covered all the unconfirmed,
+    // the next round of learn will start from committed+1.
+    //
+    if (state.__isset.learn_start_decree &&
```

Comment:
没明白`first_learn_start_decree`的用途

Reply:
> There's a problem: how can I get the max_gced_decree in learn/? To solve this we introduced first_learn_start_decree, which is the learn_start_decree in the first round of learning, stored in potential_secondary_context. It can be used to represent "the max_gced_decree under learn/".

我新加上了这段。其实就是在热备份 learn 的时候我要同时考虑 learn/ 和 plog/ 两个文件夹下的日志，如果 learn/ 目录下已经包含了我所需的日志，我就不需要再从 primary 拷贝。

那么怎么判断 learn/ 目录下有没有我所需的日志呢，我就判断 learn/ 目录的日志范围，也就是 [max_gced_decree, ~)。但是 learn/ 目录下我怎么获取 max_gced_decree 呢？我就用第一轮 learn 的起点，来作为 max_gced_decree。